test/screenshot/lib/storage.js

```diff
@@ -102,8 +109,14 @@ class Storage {
   handleUploadFailure_(uploadableFile, err) {
     const publicUrl = `${GCLOUD_STORAGE_BASE_URL}${uploadableFile.destinationAbsoluteFilePath}`;
     uploadableFile.fileContent = null; // Free up memory
-    console.error(`✗︎ FAILED to upload ${publicUrl}:`);
+    const queueIndex = uploadableFile.queueIndex;
+    const queueLength = uploadableFile.queueLength;
+    console.error(`✗︎ FAILED to upload file ${queueIndex + 1} of ${queueLength} - ${publicUrl}:`);
     console.error(err);
+    if (err.code >= 500 && err.code < 600) {
```

Comment:
This looks like it runs infinitely, too.

Reply:
Done - good catch!